packages/teleport-plugin-angular-base-component/src/utils.ts

```diff
@@ -63,25 +66,31 @@ export const generateExportAST = (
     )
   })
 
-  const classBodyAST = () => {
+  const classBodyAST = (componentUIDL: ComponentUIDL) => {
     return t.classBody([
       ...propDeclaration,
       ...propertyDecleration,
       ...dataDeclaration,
-      constructorAST(),
+      constructorAST(componentUIDL.seo),
       ...angularMethodsAST,
     ])
   }
 
   return t.exportNamedDeclaration(
-    t.classDeclaration(t.identifier(`${componentName}Component`), null, classBodyAST()),
+    t.classDeclaration(t.identifier(`${componentName}Component`), null, classBodyAST(uidl)),
     [],
     null
   )
 }
 
-const constructorAST = (t = types) => {
-  return t.classMethod('constructor', t.identifier('constructor'), [], t.blockStatement([]))
+const constructorAST = (seo: UIDLComponentSEO, t = types) => {
+  const params = []
+  if (seo) {
+    const seoBinding = t.identifier(`private titleService: Title`)
+    params.push(seoBinding)
```

Comment:
These are *typescript* specific syntaxes. Since we are using babel for constructing our AST, in most cases it works. But in cases like reserved words like `private` `public` can't be generated. Since babel parser expected only identifier in the position.

Even in the case of export class with Decorator, we were able to generate because we are taking advantage of babel's support for Flow and Typescript. Like we are combining a Flow syntax AST and normal JS AST and linking the chunks together, so the export decorator is able.